### ISSUE TITLE
`to_*` method with destructured `Self` yeilds `clippy::wrong_self_convention`

### ISSUE THREAD

**Issue by MarcusDunn** [rust-lang/rust-clippy#12593]
### Summary

de-structuring self yields `clippy::wrong_self_convention` when method is named `to_*`

### Lint Name

clippy::wrong_self_convention

### Reproducer

I tried this code:

```rust

pub struct Foo {
    n_gpu_layers: u32,
}

impl Foo {
    pub fn to_params(Self { n_gpu_layers }: Self) -> (LlamaModelParams, LlamaContextParams) {
        let llama_model_params = LlamaModelParams::default().with_n_gpu_layers(n_gpu_layers);
        let llama_context_params = LlamaContextParams::default();
        (llama_model_params, llama_context_params)
    }
}
```

I saw this happen: 

```
Methods named `to_*` normally take self by reference or self by value. Consider choosing a less ambiguous name. 
```

I expected to see this happen:
No warning.

### Version

```text
rustc 1.78.0-beta.2 (277d06bc9 2024-03-23)
binary: rustc
commit-hash: 277d06bc95c6c38a2337ccde798b2c709384bd84
commit-date: 2024-03-23
host: x86_64-unknown-linux-gnu
release: 1.78.0-beta.2
LLVM version: 18.1.2
```

### Additional Context

This yields no warning.

```rust
    pub fn to_params_2(self) -> (LlamaModelParams, LlamaContextParams) {
        let Self { n_gpu_layers } = self;
        let llama_model_params = LlamaModelParams::default().with_n_gpu_layers(n_gpu_layers);
        let llama_context_params = LlamaContextParams::default();
        (llama_model_params, llama_context_params)
    }
```

**Comment by MarcusDunn** [comment]
nvm, seems I do not know rust well enough. The two methods are different.
